src/main.h

```diff
@@ -56,6 +56,8 @@ static const unsigned int DEFAULT_BLOCK_PRIORITY_SIZE = DEFAULT_BLOCK_MAX_SIZE /
 static const bool DEFAULT_ALERTS = true;
 /** Minimum alert priority for enabling safe mode. */
 static const int ALERT_PRIORITY_SAFE_MODE = 4000;
+/** Maximum reorg length we will accept before we shut down and alert the user. */
+static const unsigned int MAX_REORG_LENGTH = COINBASE_MATURITY - 1;
```

Comment:
Selected this because for current nodes, a larger reorg would crash their nodes. It has the additional economic side-effect of ensuring that by default, nodes do not accept re-orgs that delete currently-spendable coinbase.

Reply:
Why does a larger reorg cause a crash? (Think it's a good limit anyways)

Reply:
See #1302. tl;dr if a rollback causes the note witness cache to become empty, an assertion is triggered.